src/robot/variables/search.py

```diff
@@ -114,13 +114,15 @@ def is_list_variable(self):
     def is_dict_variable(self):
         return self.identifier == '&' and self.is_variable()
 
-    def is_assign(self, allow_assign_mark=False, allow_nested=False):
+    def is_assign(self,
+                  allow_assign_mark=False, allow_nested=False, allow_items=False):
         if allow_assign_mark and self.string.endswith('='):
             match = search_variable(self.string[:-1].rstrip(), ignore_errors=True)
-            return match.is_assign()
+            return match.is_assign(allow_items=allow_items)
         return (self.is_variable()
                 and self.identifier in '$@&'
-                and not self.items
+                and not (self.identifier == '&' and self.items)
+                and (allow_items or not self.items)
```

Comment:
We probably should allow item assignment only with `$` variables. This code already prohibits it with `&` variables, but I don't think it makes sense with `@` variables either.

Reply:
By allowing the `@`, we can actually use syntax like this one:
```robot
${mylist}=        Create List      ${1}    ${2}    ${3}
${mylist}[0]      @{mylist}[1]=    Set Variable     ${{ [9, 8, 7, 6] }}
Should Be Equal    ${mylist}    ${{ [9, [8, 7, 6], 3] }}
```
which is the equivalent of the Python:
```python3
mylist = [1, 2, 3]
mylist[0], *mylist[1] = [9, 8, 7, 6]
mylist   # [9, [8, 7, 6], 3]
```

It's tested in `atest/testdata/variables/return_values.py` by the tests:
* `Multiple item assigns to scalars and list`
* `Multiple item assigns to scalars and list slice`
* `Multiple item assigns to list only`

Do you think we can keep this feature?

Reply:
I didn't actually know `mylist[0], *mylist[1] = [9, 8, 7, 6]` is valid in Python. That's pretty obscure and I'm not sure does it make sense for us to support it. Do you have examples where this syntax would be useful in Python or with Robot?

It's great that you added tests for this functionality. It makes me want to just merge this as-is, but I need to think about this a bit more. I think I'll ask from other on the #devel channel on our Slack as well.

Reply:
TBH, I don't remember seeing this syntax anywhere before too...
I guess it could be useful to unpack a returned iterable into dict with one-liner like so:
```python
mydict["name"], *mydict["args"] = get_sysargv()
```
But it's a bit far fetched...


For your other question, until today we got the following result:
```robot
${x} =       Create List    a    b    c
@{x}[0] =    Create List    x    y    x   
# ${x} => [['x', 'y', 'z'], 'b', 'c']
# @{x}[0] is equivalent to ${x}[0]
```
I agree with you it doesn't make much sense.
**So I added today the logic in the `assigner.py` file to forbid this. User will get the error "Cannot assign a list variable with items."**


I can think of two options:
1. We keep today's logic and allow syntax like below:
```robot
@{x}[a]  ${y}[b]  ${z}[c]=   ...
```

2. We prevent this syntax. Then I can remove the logic to handle this and add a simple condition in `search.py` to prevent any `@` variable from using items => _**_least confusing solution_**_.

Reply:
I asked about the
```
    ${x}    @{y}[1] =    Keyword
```
syntax on Slack and the only comment I got was very positive. I can also see it as a logic extension to the syntax we otherwise support and, although it gets pretty complicated, see no real harm with it. Let's add it.

Reply:
If we allow `@{x}[0]` syntax when assigning multiple variables, I believe we should also allow
```
@{x}[0] =    Keyword 1
&{y}[key] =    Keyword 2
```
They should work exactly like
```
${x}[0] =    Keyword 1
${y}[key] =    Keyword 2
```
except that
- the first one with `@` should require the return value to be a list-like and convert it to a list, and
- the second one with `&` should require the return value to be a dict-like and convert it to a `DotDict`.
 
The above is exactly the difference between assigning
```
${x} =    Keyword 1
${y} =    Keyword 2
```
and
```
@{x} =    Keyword 1
&{y} =    Keyword 2
```
With this design the behavior ought to be consistent.

Reply:
The commit 53aa8ca adds support for the syntax:
```robot
@{x}[0] =      Keyword 1
&{y}[key] =    Keyword 2
```
They should now be consistent with the behavior of:
```robot
@{x} =    Keyword 1
&{y} =    Keyword 2
```
Please see the 4 last test cases in `return_values.robot`. Let me know if something must be improved 🙏.